src/spdx_matcher/__init__.py

```diff
@@ -198,142 +201,151 @@ def normalize(license_text, remove_sections=REMOVE_FINGERPRINT):
+            "text_length": len(license["licenseText"]),
+            "matchConfidence": license["matchConfidence"],
+            "isDeprecatedLicenseId": license["isDeprecatedLicenseId"],
+        }
+
+
+def _process_exception(exception, match_cache):
+    if "detailsUrl" in exception:
+        json_detail_data = _fetch_json(exception["detailsUrl"])
+        if "licenseExceptionTemplate" in json_detail_data:
+            json_detail_data["regexpForMatch"] = _convert_template_to_regexp(
+                json_detail_data["licenseExceptionTemplate"]
+            )
+        if "licenseExceptionText" in json_detail_data:
+            if "regexpForMatch" in json_detail_data:
+                start_time = time.time()
+                match, license_data, full_match = _license_regexps_match(
+                    json_detail_data["regexpForMatch"],
+                    json_detail_data["licenseExceptionText"],
+                    fast_exit=False,
+                )
+                execution_time = time.time() - start_time
+                json_detail_data["matchCost"] = execution_time
+                json_detail_data["matchConfidence"] = match
+                if match < 0.99:
+                    print(
+                        f"Unable to match {match} full_match:{full_match}  exmplar text "
+                        f"to regexp for exception "
+                        f"{json_detail_data['licenseExceptionId']}  "
+                        f"license_data {json.dumps(license_data)} time {execution_time}"
+                    )
+                else:
+                    print(
+                        f"Success to match and full_match:{full_match} exmplar text to "
+                        f"regexp for exception {json_detail_data['licenseExceptionId']} "
+                        f"license_data {json.dumps(license_data)} time {execution_time}"
+                    )
+        for k in json_detail_data:
+            if k not in exception:
+                exception[k] = json_detail_data[k]
+        match_cache["exceptions"][exception["licenseExceptionId"]] = {
+            "name": exception["name"],
+            "regexpForMatch": exception["regexpForMatch"],
+            "matchCost": exception["matchCost"]
+            if "matchCost" in exception
+            else 100.0,
+            "text_length": len(exception["licenseExceptionText"]),
+            "matchConfidence": exception["matchConfidence"],
+            "isDeprecatedLicenseId": exception["isDeprecatedLicenseId"],
+        }
+
+
+def cache_builder():
+    match_cache = {"licenses": {}, "exceptions": {}}
+    base_url = "https://spdx.org/licenses"
+
+    # Fetch licenses.json
+    licenses_data = _fetch_json(f"{base_url}/licenses.json")
+    exceptions_data = _fetch_json(f"{base_url}/exceptions.json")
+
+    # Process licenses and exceptions using ThreadPoolExecutor
+    with ThreadPoolExecutor(max_workers=DEFAULT_CACHE_THREAD_POOL_WORKERS) as executor:
```

Comment:
Use ThreadPoolExecutor build the cache data fast 🚀 , currently can finish within 2min